[WPE][GTK] Document remote web inspector usage

aperezdc opened the pull request:
#### 98b62abac986c1500a00f7412c22394bd660cc10
<pre>
[WPE][GTK] Document remote web inspector usage
<a href="https://bugs.webkit.org/show_bug.cgi?id=286799">https://bugs.webkit.org/show_bug.cgi?id=286799</a>

Reviewed by Carlos Garcia Campos.

Add a new Markdown page for the GTK and WPE port documentation
explaining how to run the remote Web Inspector server, and how
to connect to it from another browser.

* Source/WebKit/PlatformGTK.cmake: List new Markdown template.
* Source/WebKit/PlatformWPE.cmake: Ditto.
* Source/WebKit/glib/environment-variables.md.in: Add variables related
  to the remote inspector server.
* Source/WebKit/glib/remote-inspector.md.in: Added.
* Source/WebKit/gtk/gtk3-webkitgtk.toml.in: List new Markdown page.
* Source/WebKit/gtk/gtk4-webkitgtk.toml.in: Ditto.
* Source/WebKit/wpe/wpewebkit.toml.in: Ditto.

Canonical link: <a href="https://commits.webkit.org/289603@main">https://commits.webkit.org/289603@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e1f1025778874a48df41e9ff4ef77258870c93eb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/87509 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/7022 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/41884 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/92371 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/38249 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/7404 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/15191 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/67587 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25332 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/90511 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/5644 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/79190 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/47939 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/5430 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/37364 "Built successfully") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/34448 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/94256 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/14674 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/10764 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/76417 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/14928 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/75045 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/75644 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/20021 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/18445 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/7621 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13627 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/14692 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/19987 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/14435 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/17879 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/16218 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/glib/remote-inspector.md.in:
```diff
@@ -0,0 +1,64 @@
+
+#if WPE
+
+Note WPE WebKit only supports using the Web Inspector remotely, as embedded
+devices often lack input devices, have smaller screens, use custom windowing
+systems, or may even be headless (no screen at all).
```
This is not accurate, WPE with the new api supports the local inspector.

aperezdc replied:
Good point, let's replace `s/only/always/`, and add a little note:

```
Note WPE WebKit always supports using the Web Inspector remotely, as embedded
devices often lack input devices, have smaller screens, use custom windowing
systems, or may even be headless (no screen at all). Using the Web Inspector
locally is also possible when built with `ENABLE_WPE_PLATFORM` enabled.
```

WDYT?

carlosgcampos replied:
It's not a build time decision, but after thinking again maybe it's better not to mention the new api, since it's not officially released yet...

aperezdc replied:
Right, then let's land it as-is, and amend the documentation later on when the new API is deemed stable.

aperezdc committed:
[WPE][GTK] Document remote web inspector usage
https://bugs.webkit.org/show_bug.cgi?id=286799

Reviewed by Carlos Garcia Campos.

Add a new Markdown page for the GTK and WPE port documentation
explaining how to run the remote Web Inspector server, and how
to connect to it from another browser.

* Source/WebKit/PlatformGTK.cmake: List new Markdown template.
* Source/WebKit/PlatformWPE.cmake: Ditto.
* Source/WebKit/glib/environment-variables.md.in: Add variables related
  to the remote inspector server.
* Source/WebKit/glib/remote-inspector.md.in: Added.
* Source/WebKit/gtk/gtk3-webkitgtk.toml.in: List new Markdown page.
* Source/WebKit/gtk/gtk4-webkitgtk.toml.in: Ditto.
* Source/WebKit/wpe/wpewebkit.toml.in: Ditto.

Canonical link: https://commits.webkit.org/289603@main

spenap commented on Source/WebKit/glib/remote-inspector.md.in:
````diff
@@ -0,0 +1,64 @@
+```
+
+The Web Inspector is an application written using Web technologies itself,
+and it may be loaded on non-WebKit browsers. If enabled, the [inspector
````
> it may be loaded on non-WebKit browsers.

I missed when this landed, so sorry my review is late. 

A while ago, I tried unsuccessfully to use remote web inspector from a Firefox browser on my host to inspect a WPE browser on a Raspberry Pi. It worked fine when I used Epiphany instead.

I was using `WEBKIT_INSPECTOR_HTTP_SERVER`

aperezdc replied:
I think most of the time Chromium tends to work better because it's what is used to test the HTTP RWI; we typically don't have much time to spare to ensure Firefox would always work.

spenap commented on Source/WebKit/glib/environment-variables.md.in:
```diff
@@ -38,3 +38,14 @@ You can ask WebKit to try to use a format instead the one that is recommended by
 #endif
+
+
+## Remote Web Inspector Variables
+
+- `WEBKIT_INSPECTOR_SERVER`: The address and port in `ip:port` format (e.g.
+  `127.0.0.1:10000`) where the remote inspector service will listen for
+  connections.
+- `WEBKIT_INSPECTOR_HTTP_SERVER`: The address and port in `ip:port` format
+  where the remote inspector service will listen for HTTP connections.
+
+See [Remote Web Inspector](remote-inspector.html) for more details.
```
Could this be clarified further? I.e. does `WEBKIT_INSPECTOR_SERVER` listen for non-HTTP connections? What kind of connections are these? 

Is there a recommendation to use one vs the other? If so, we could probably say

When using non-WebKit-based browsers, it is recommended to use `WEBKIT_INSPECTOR_HTTP_SERVER`. On WebKit-based browsers, `WEBKIT_INSPECTOR_SERVER` is enough. See [Remote Web Inspector](...) for more details

spenap reviewed (commented):
I am going through emails I missed and found this. Where is this documentation published to? Would it make sense to copy this documentation over to WebKit Documentation?
